### PR TITLE
fabtests/efa: Improve MR exhaustion test

### PR DESCRIPTION
**Pull request by sunkuamzn**
Some improvements to the MR exhaustion test
* Add a simple test for the unexpected message path
* Get device MR limit by querying the device instead of hardcoding the value
* Close ibv device after use

### CODE REVIEW

File: fabtests/prov/efa/src/efa_exhaust_mr_reg_common.c [hunk 1]
```diff
@@ -2,8 +2,7 @@
 
 #include "efa_exhaust_mr_reg_common.h"
 
-int ft_efa_setup_ibv_pd(struct ibv_pd **pd)
-{
+struct ibv_context *ft_efa_open_ibv_device() {
```
Comment (wenduwan) [inline fabtests/prov/efa/src/efa_exhaust_mr_reg_common.c:5]: I think the canonical signature is `int ft_efa_open_ibv_device(struct ibv_context *ctx)`

For nil-parameter functions we should do `function(void)`

**Review by a-szegel — APPROVED** [review]

**Comment by a-szegel** [comment]
Make sure to grab Wenduo's comment, also looks like there are some real fabtest failures here.

**Comment by sunkuamzn** [comment]
bot:aws:retest

File: fabtests/prov/efa/src/efa_exhaust_mr_reg_common.c (hunk 1 above)
Comment (sunkuamzn) [reply fabtests/prov/efa/src/efa_exhaust_mr_reg_common.c]: fixed, thanks!

**Review by wenduwan — APPROVED** [review]

**Comment by sunkuamzn** [comment]
Fixed fabtest failures. CI passing now.

**Review by wenduwan — APPROVED** [review]

**Pull request merged**